app/src/main/java/org/wikipedia/main/MainActivity.kt

```diff
@@ -230,7 +237,13 @@ class MainActivity : SingleFragmentActivity<MainFragment>(), MainFragment.Callba
         super.onResume()
 
         // start cronet here to prevent exception from starting a service when out of focus
-        if (CronetNetworking.cronetEngine() != null) {
+        if (Prefs.isInitialOnboardingEnabled) {
+            // TODO: onCreate also checks the following before onboarding, is that necessary here?
+            // savedInstanceState == null && !intent.hasExtra(Constants.INTENT_EXTRA_IMPORT_READING_LISTS
+            Log.d(TAG, "user is likely doing onboarding, don't try to start envoy")
+        } else if (envoyUnused) {
+            Log.d(TAG, "direct connection previously worked, don't try to start envoy")
+        } else if (CronetNetworking.cronetEngine() != null) {
```

Comment:
Will it try envoy after `isInitialOnboardingEnabled` is done?

Reply:
since it's triggered in onResume it will try again when the user finishes with onboarding and MainActivity gets focus again.  i can spend some additional time testing edge cases though (ie: close app during onboarding).